.eslintrc.js

```diff
@@ -9,280 +9,161 @@ const dictionary = require('./.eslint-dictionary.json');
+        ],
+        minLength: 4,
       },
     ],
-    '@typescript-eslint/explicit-function-return-type': ['error', { allowExpressions: true }],
-    '@typescript-eslint/no-explicit-any': 'error',
-    '@typescript-eslint/method-signature-style': ['error', 'property'],
-
-    // Some ESLint rules conflict with the corresponding TS rule. These ESLint rules are turned off in favor of the corresponding TS rules
-    'no-invalid-this': 'off',
-    '@typescript-eslint/no-invalid-this': 'error',
-    'no-unused-vars': 'off',
-    '@typescript-eslint/no-unused-vars': ['error', { vars: 'all', args: 'all', argsIgnorePattern: '^_{2,}[A-Za-z0-9]*$' }],
-    'no-shadow': 'off',
-    '@typescript-eslint/no-shadow': 'error',
-    'no-useless-constructor': 'off',
-    '@typescript-eslint/no-useless-constructor': 'error',
-
-    // Import Rules
-    // Extends recommended rules here: https://github.com/import-js/eslint-plugin-import/blob/6c957e7df178d1b81d01cf219d62ba91b4e6d9e8/config/recommended.js
-    'import/no-dynamic-require': 'error',
-    'import/newline-after-import': 'error',
-    'import/no-cycle': 'error',
-    'import/order': 'error',
-    'import/first': 'error',
-    'import/no-extraneous-dependencies': 'error',
-    'import/no-useless-path-segments': 'error',
-    'import/extensions': 'off',
-    'import/prefer-default-export': 'off',
-
-    // JSDoc Rules
-    // Docs here: https://www.npmjs.com/package/eslint-plugin-jsdoc
-    'jsdoc/require-jsdoc': ['error', {
-      publicOnly: true,
-      require: {
-        ClassDeclaration: true,
-        ArrowFunctionExpression: true,
-      },
-      contexts: [
-        'MethodDefinition:not([accessibility=/(private|protected)/]) > FunctionExpression', // Require JSDoc on public methods
-        'TSInterfaceDeclaration',
-        'TSTypeAliasDeclaration',
-        'TSEnumDeclaration',
-      ],
-      checkConstructors: false
-    }],
-    'jsdoc/require-description': ['error', { contexts: ['any'] }
-    ],
-    'jsdoc/require-param': 'off',
-    'jsdoc/require-param-description': 'error',
-    'jsdoc/require-returns': 'off',
-    'jsdoc/require-returns-description': 'error',
-    'jsdoc/check-param-names': 'error',
-
-    // ESLint Rules
-    // These rules override the AirBnb rules here: https://github.com/airbnb/javascript
-    // as well as the recommended ESLint rules here: https://eslint.org/docs/rules/
-
-    // this is the same as the AirBnb rule, but with length of 140 instead of 100
-    'max-len': ['error', 140, 2, {
-      ignoreUrls: true,
-      ignoreComments: false,
-      ignoreRegExpLiterals: true,
-      ignoreStrings: true,
-      ignoreTemplateLiterals: true,
-    }],
-    'lines-between-class-members': ['error', 'always', { exceptAfterSingleLine: true }],
-    'max-classes-per-file': 'error',
-    'no-lonely-if': 'error',
-    'no-unneeded-ternary': 'error',
-    'no-use-before-define': 'off',
-    'consistent-return': 'error',
```

Comment:
I think this rule is valuable

Reply:
Currently, there are 90 violations of the rule in the codebase. The original reason for disabling this rule in this iteration is that fixing the rule could potentially cause unexpected behavior in the codebase. That said, in most cases, a fix is relatively straightforward. It may be worth including the rule, fixing what is easily fixable, and then deciding a path forward on the remaining violations (potentially including locally disabling the rule)